src/libs/ReportUtils.js

```diff
@@ -1426,28 +1426,39 @@ function buildOptimisticIOUReportAction(type, amount, currency, comment, partici
     };
 }
 
-function buildOptimisticReportPreview(reportID, iouReportID, payeeAccountID) {
-    return {
+function buildOptimisticReportPreview(chatReport, iouReport, existingReportPreviewAction, justPaidOutstandingIOU) {
+    const reportPreviewAction = existingReportPreviewAction || {
         reportActionID: NumberUtils.rand64(),
-        reportID,
-        created: DateUtils.getDBTime(),
+        reportID: chatReport.reportID,
         actionName: CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW,
         pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
-        accountID: payeeAccountID,
-        message: [
-            {
-                html: '',
-                text: '',
-                isEdited: false,
-                type: CONST.REPORT.MESSAGE.TYPE.COMMENT,
-            },
-        ],
         originalMessage: {
-            linkedReportID: iouReportID,
+            linkedReportID: iouReport.reportID,
         },
-        actorEmail: currentUserEmail,
-        actorAccountID: currentUserAccountID,
     };
```

Comment:
Given that this only occurs if `existingReportPreviewAction` is passed, I wonder if it would be better to split this logic into two functions. Or simply move the 'use existing report preview action' condition to the place where the function is invoked? It prevents the need to pass something that is only used once within the function.

Reply:
fixed.